src/utils/gauges.ts

```diff
@@ -372,3 +430,171 @@ export function areGaugesActive(chainId?: ChainId): boolean {
     IS_VESDL_LIVE
   )
 }
+
+async function getGaugeCount(
+  childGaugeFactory: ChildGaugeFactory | null,
+  gaugeController: GaugeController | null,
+): Promise<BigNumber> {
+  if (childGaugeFactory) {
+    return await childGaugeFactory.get_gauge_count()
+  }
+
+  if (gaugeController) {
+    return await gaugeController.n_gauges()
+  }
```

Comment:
Heads up: on a side chain `childGaugeFactory.get_gauge_count()` will return the number of ChildGauges existing on the side chain. Including permissionlessly deployed ChildGauges. (any user can trigger a deployment of a childgauge via `ChildGaugeFactory` but the corresponding root gauge will not be included in gauge controller until its been added by our multisig)

But on mainnet, `gaugeController.n_gauges()` returns number of RootGauges + LiquidityGaugeV5s that are approved by Saddle DAO.

Reply:
@penandlim if that's the case, is there a flag or some call we can make to ensure that we only display gauges that have been added by our multisig?

Reply:
Unfortunately not on side chains :/

The official list of gauge addresses that are eligible to be voted by veSDL holders is maintained in GaugeController contract on eth mainnet.

So would have to rely on an offchain or cross chain solution.

Off chain API would be the best imo.

Reply:
Safe to ignore this as ChildGaugeFactory has unique LPToken <> Gauge mapping

Can be overriden by the multisig in case malicious user.

Although we cannot check whether this child gauge is added to the gauge controller yet, we can safely assume it is if it's SDL emission is >0. 

When SDL emission is 0, it would mean no one voted for the corresponding root gauge or that the root gauge is not added to the gauge controller yet.